ooiservices/app/uframe/controller.py

```diff
@@ -310,3 +313,105 @@ def get_data(stream, instrument,field):
 
     #return jsonify(**resp_data)
     return resp_data
+
+#@api.route('/get_profiles/<string:reference_designator>/<string:stream_name>')
+#def get_profiles(reference_designator, stream_name):@api.route('/get_profiles')
+def get_profiles():
+    #data = get_data(reference_designator, stream_name)
+    data = requests.get('http://localhost:12570/sensor/user/inv/ctdpf_ckl_wfp_instrument/CP02PMUO-WFP01-03-CTDPFK000')
+    data = data.json() 
+# Note: assumes data has depth and time is ordinal
+# Need to add assertions and try and exceptions to check data
+
+    time = []
+    depth = []
+    preferred_timestamp = data[0]['preferred_timestamp']
+        #preferred_timestamp = preferred_timestamp * 2208988800
+
+    
+    for row in data:
+        depth.append(int(row['pressure']))
+        time.append(float(row[preferred_timestamp]))
+    
+    matrix = np.column_stack((time, depth))
+    tz = matrix
+    origTz = tz
+    INT = 10
+    print len(tz)#this length must equal profile_list length
+
+    maxi = np.amax(tz[:,0])
+    mini =np.amin(tz[:,0])
+    #getting a range from min to max time with 10 seconds or milliseconds. I have no idea. 
+    ts =(np.arange(np.amin(tz[:,0]), np.amax(tz[:,0]), INT)).T
+
+    #interpolation adds additional points on the line within f(t), f(t+1)  time is a function of depth
+    itz = np.interp(ts,tz[:,0],tz[:,1])
+
+
+    newtz= np.column_stack((ts, itz))
+    # 5 unit moving average 
+    WINDOW = 5
+    weights = np.repeat(1.0, WINDOW)/ WINDOW
+    ma =np.convolve(newtz[:,1], weights)[WINDOW-1:-(WINDOW-1)]
+    #take the diff and change negatives to -1 and postives to 1
+    dZ = np.sign(np.diff(ma))
+
+    # repeat for second derivative  
+    dZ = np.convolve(dZ, weights)[WINDOW-1:-(WINDOW-1)]
+    dZ = np.sign(dZ)
+
+    r0=1
+    r1 = len(dZ)+1
+    dZero = np.diff(dZ)
+
+    start =[] 
+    stop =[]
+    #find where the slope changes 
+    dr = [start.append(i)  for (i, val) in enumerate(dZero) if val !=0]
+
+    for i in range(len(start)-1):
+        stop.append(start[i+1])
+    stop.append(start[0])
+    start_stop = np.column_stack((start, stop))
+
+    start_times = np.take(newtz[:,0], start)
+
+    stop_times = np.take(newtz[:,0], stop)
+
+    start_times = start_times - INT*2
+    stop_times = stop_times + INT*2
+
+
+    depth_profiles=[]
+    for i in range(len(start_times)):
+        profile_id=i
+        proInds= origTz[(origTz[:,0] >= start_times[i]) & (origTz[:,0] <= stop_times[i])]
+        value = proInds.shape[0]
+        z = np.full((value,1), profile_id)
+        pro = np.append(proInds, z, axis=1)
+        depth_profiles.append(pro)
+    depth_profiles = np.concatenate(depth_profiles)
+# I NEED to CHECK FOR DUPLICATE TIMES !!!!! NOT YET DONE!!!!
```

Comment:
It would be better to make this something easier to find, such as 

#@TODO Check for duplicate Items